### PR TITLE
Support structured logging

### PR DESCRIPTION
**Pull request by bells17**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

I've added JSON logging and changed to use structured logging (e.g., klog.InfoS) in this PR.
With this change, it is now possible to output the logs of the external-health-monitor container in JSON format.
Running the container with the `--logging-format=json` option will output the logs in JSON format.

In addition, I've modified the log messages based on the following guideline:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#remove-string-formatting-from-log-message

I’ve update the klog functions in use according to the guidelines provided below, and I've confirmed that they pass the [logcheck](https://github.com/kubernetes-sigs/logtools/tree/main/logcheck) tests:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#change-log-functions

```
$ logcheck cmd/csi-external-health-monitor-controller/main.go
$ logcheck ./pkg/controller/
$ logcheck ./pkg/csi-handler/
$ logcheck ./pkg/csi-handler/
$ logcheck ./pkg/mock/
$ logcheck ./pkg/util/
```

related PR: https://github.com/kubernetes-csi/livenessprobe/pull/202

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

```
# The logs below were outputted because the program was running on MacOS.
$ go run cmd/csi-external-health-monitor-controller/main.go --logging-format=json
{"ts":1692983306225.366,"caller":"csi-external-health-monitor-controller/main.go:103","msg":"Version","v":0,"version":"unknown"}
{"ts":1692983306225.402,"caller":"csi-external-health-monitor-controller/main.go:117","msg":"Failed to build a Kubernetes config","err":"unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"}
exit status 1

$ go run cmd/csi-external-health-monitor-controller/main.go
I0826 02:08:31.658543   19448 main.go:103] "Version" version="unknown"
E0826 02:08:31.658664   19448 main.go:117] "Failed to build a Kubernetes config" err="unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"
exit status 1
```

**Does this PR introduce a user-facing change?**:

```release-note
Added support structured logging
```

### CODE REVIEW

**Comment by xing-yang** [comment]
Hi @bells17, it looks like you are not a member yet.  Please feel free to open an issue [here](https://github.com/kubernetes/org/issues) and apply for membership in kubernetes-csi org.

**Comment by bells17** [comment]
I apologize for the oversight. I've noticed that the klog functions in use are not aligned with the guidelines below, so I will be changing the status of the Pull Request to Draft for now:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#change-log-functions

**Comment by bells17** [comment]
The changes are done, so I've changed the PR status back to 'ready'.

File: cmd/csi-external-health-monitor-controller/main.go [hunk 1]
```diff
@@ -81,19 +85,26 @@ var (
 )
 
 func main() {
-	klog.InitFlags(nil)
-	flag.Set("logtostderr", "true")
+	fg := featuregate.NewFeatureGate()
```
Comment (xing-yang) [inline cmd/csi-external-health-monitor-controller/main.go:88]: What is this feature gate for?

Comment (bells17) [reply cmd/csi-external-health-monitor-controller/main.go:88]: This feature gate is required for the execution of the `logsapi.ValidateAndApply` function.

https://github.com/kubernetes-csi/external-health-monitor/pull/208/files#diff-6d86851f5b7475589c681f792232f791711c4a1aad3073c0caf817f6f9744becR94

File: pkg/controller/node_watcher.go [hunk 2]
```diff
@@ -172,7 +171,7 @@ func (watcher *NodeWatcher) WatchNodes() {
 			return false
 		}
 		if !errors.IsNotFound(err) {
-			klog.V(2).Infof("error getting node %q from informer: %v", key, err)
+			klog.V(2).ErrorS(err, "Error getting node from informer", "node", key)
```
Comment (xing-yang) [inline pkg/controller/node_watcher.go:174]: Can you keep this at InfoS?

Comment (bells17) [reply pkg/controller/node_watcher.go]: Sorry for the mistake. I've updated it to use InfoS.

Comment (bells17) [reply pkg/controller/node_watcher.go]: @xing-yang Sorry for the mistake. I've updated it to use InfoS.

**Pull request merged**